ConsoleLogger logs to stderr

Following the intention that logging should behave similar than `console.log`, it is suprising to see that the `ConsoleLogger` uses `console.error()` to log the messages; which effectively logs to `stderr`.

Especially for CLI programs, where CLI output might be interspersed with log messages, logging to different streams might be a problem.

Of course, when building a pipe-able program, one might explicitely only log to stderr.

## Suggestion:

- make it configurable in the `ConsoleLogger`, which stream to use.
- default to `stdout`

ps: I would use stdout/stderr directly, and not using `console.[log|error]`. it is unlikely ATM that helix-log is used in the browser (also, since it tries to load the package.json :-).

> Not sure the out of order problem would actually happen; 

for example in openwhisk actions, where both streams are forwarded to the same pipe.
since the pipes are async, the order is not guaranteed. 

see the following 2 requests to the identical action: once stdout is added first, once after. also see the mangled timestamps:

```
$ wsk activation logs 078b05ef00d046b18b05ef00d036b1f3
2019-09-24T04:30:08.985Z       stdout: configured coralogix logger.
2019-09-24T04:30:08.986Z       stdout: configured papertrail logger.
2019-09-24T04:30:08.985Z       stderr: [ERROR] this is a test
2019-09-24T04:30:08.988Z       stderr: [INFO] GET /ping 200 - 0.91 ms
```

```
$ wsk activation logs 8d2f126a6766456aaf126a6766656a42
2019-09-24T04:30:43.186Z       stderr: [ERROR] this is a test
2019-09-24T04:30:43.190Z       stderr: [INFO] GET /ping 200 - 0.76 ms
2019-09-24T04:30:43.186Z       stdout: configured coralogix logger.
2019-09-24T04:30:43.188Z       stdout: configured papertrail logger.
```

--- a/README.md
+++ b/README.md
@@ -458,6 +458,7 @@ all non-strings.
     * [.level](#ConsoleLogger+level) : <code>string</code>
     * [.formatter](#ConsoleLogger+formatter) : [<code>MessageFormatter</code>](#MessageFormatter)
     * [.fmtOpts](#ConsoleLogger+fmtOpts) : <code>object</code>
+    * [.stream](#ConsoleLogger+stream) : <code>Writable</code>
     * [.log(msg, opts)](#ConsoleLogger+log)
 
 <a name="new_ConsoleLogger_new"></a>
@@ -468,6 +469,7 @@ all non-strings.
 | --- | --- | --- | --- |
 | opts | <code>Object</code> |  | – Configuration object. All other options are forwarded to the formatter. |
 | [opts.level] | <code>string</code> | <code>&quot;info&quot;</code> | The minimum log level |
+| [opts.stream] | <code>Writable</code> | <code>console._stderr</code> | A writable stream to log to. |
 | [opts.formatter] | [<code>MessageFormatter</code>](#MessageFormatter) | <code>messageFormatConsole</code> | A formatter producing strings |
 
 <a name="ConsoleLogger+level"></a>
@@ -490,6 +492,12 @@ Feel free to mutate or exchange.
 ### consoleLogger.fmtOpts : <code>object</code>
 Options that will be passed to the formatter;
 Feel free to mutate or exchange.
+
+**Kind**: instance property of [<code>ConsoleLogger</code>](#ConsoleLogger)  
+<a name="ConsoleLogger+stream"></a>
+
+### consoleLogger.stream : <code>Writable</code>
+Writable stream that is used to write the log messages to.
 
 **Kind**: instance property of [<code>ConsoleLogger</code>](#ConsoleLogger)  
 <a name="ConsoleLogger+log"></a>

--- a/src/log.js
+++ b/src/log.js
@@ -349,6 +349,7 @@ const messageFormatJson = (msg, opts) => {
  * @class
  * @param {Object} opts – Configuration object. All other options are forwarded to the formatter.
  * @param {string} [opts.level=info] The minimum log level
+ * @param {Writable} [opts.stream=console._stderr] A writable stream to log to.
  * @param {MessageFormatter} [opts.formatter=messageFormatConsole] A formatter producing strings
  */
 class ConsoleLogger {
@@ -374,11 +375,21 @@ class ConsoleLogger {
    * @member {object} fmtOpts
    */
 
+  /**
+   * Writable stream that is used to write the log messages to.
+   * @memberOf ConsoleLogger#
+   * @member {Writable} stream
+   */
+
   /* istanbul ignore next */
   constructor(opts = {}) {
     /* istanbul ignore next */
-    const { level = 'info', formatter = messageFormatConsole, ...fmtOpts } = opts;
-    assign(this, { level, formatter, fmtOpts });
+    const {
+      level = 'info', stream = console._stderr, formatter = messageFormatConsole, ...fmtOpts
+    } = opts;
+    assign(this, {
+      level, formatter, fmtOpts, stream,
+    });
   }
 
   /* istanbul ignore next */
@@ -388,9 +399,7 @@ class ConsoleLogger {
     /* istanbul ignore next */
     const { level = 'info' } = opts || {};
     if (numericLogLevel(level) <= numericLogLevel(this.level)) {
-      // Logs should go to stderr; this is only correct in node;
-      // in the browser we should use console.log
-      console.error(this.formatter(msg, { ...this.fmtOpts, level }));
+      this.stream.write(`${this.formatter(msg, { ...this.fmtOpts, level })}\n`);
     }
   }
 }

--- a/test/log.test.js
+++ b/test/log.test.js
@@ -274,6 +274,10 @@ it('ConsoleLogger', async () => {
 
   out = await recordStderr(() => testLogger(new ConsoleLogger({ level: 'fatal' })));
   assert.strictEqual(out, logOutputFatalConsole);
+
+  const ss = new StringStream();
+  testLogger(new ConsoleLogger({ level: 'fatal', stream: ss }));
+  assert.strictEqual(ss.extract(), logOutputFatalConsole);
 });
 
 it('FileLogger', async () => {

